iamctl/pkg/applications/import.go

```diff
@@ -103,39 +106,64 @@ func importApp(importFilePath string, isUpdate bool) error {
 	modifiedFileData := utils.ReplaceKeywords(string(fileBytes), appKeywordMapping)
 
 	if isUpdate {
-		log.Println("Updating application: " + fileInfo.ResourceName)
-		err = utils.SendUpdateRequest("", importFilePath, modifiedFileData, utils.APPLICATIONS)
-	} else {
-		log.Println("Creating new application: " + fileInfo.ResourceName)
-		err = utils.SendImportRequest(importFilePath, modifiedFileData, utils.APPLICATIONS)
+		return updateApplication(importFilePath, modifiedFileData, fileInfo)
+	}
+	return importApplication(importFilePath, modifiedFileData, fileInfo)
+}
+
+func updateApplication(importFilePath string, modifiedFileData string, fileInfo utils.FileInfo) error {
+
+	log.Println("Updating application: " + fileInfo.ResourceName)
+	err := utils.SendUpdateRequest("", importFilePath, modifiedFileData, utils.APPLICATIONS)
+	if err != nil {
+		utils.UpdateFailureSummary(utils.APPLICATIONS, fileInfo.ResourceName)
+		return fmt.Errorf("Error when updating application: %s", err)
 	}
+	utils.UpdateSuccessSummary(utils.APPLICATIONS, utils.UPDATE)
+	log.Println("Application updated successfully.")
+	return nil
+}
+
+func importApplication(importFilePath string, modifiedFileData string, fileInfo utils.FileInfo) error {
+
+	log.Println("Creating new application: " + fileInfo.ResourceName)
+	err := utils.SendImportRequest(importFilePath, modifiedFileData, utils.APPLICATIONS)
 	if err != nil {
-		return fmt.Errorf("error when importing application: %s", err)
+		utils.UpdateFailureSummary(utils.APPLICATIONS, fileInfo.ResourceName)
+		return fmt.Errorf("Error when importing application: %s", err)
 	}
+
+	if authenticated, err := isAuthenticationApp(modifiedFileData); err != nil {
+		fmt.Println("error occurred:", err)
+	} else if authenticated {
+		utils.AddNewSecretApplication(fileInfo.ResourceName)
```

Comment:
Let's change the method name. This implies adding a new secret app. Add a comment here as well.

Reply:
Have we checked the scenario where an application exported with secrets is imported? We are still creating a new secret for that too during import, right? Shall we recheck that behaviour?

Reply:
Changed the method name to `IndicateNewSecretGenerated` with the string array named `SecretGeneratedApplications`

Reply:
Only when creating a new Oauth app without giving a secret the indicator is added.